app/controllers/application_controller.rb

```diff
@@ -4,10 +4,23 @@
 class ApplicationController < ActionController::Base
   include Clearance::Controller
   protect_from_forgery with: :exception
+  before_action :check_company_subdomain
 
   private
 
   def current_company
     current_user.company
   end
+
+  def check_company_subdomain
+    subdomain = request.subdomain
+    return if subdomain.empty? || subdomain == 'www'
+    matching_company = Company.find_by domain: subdomain
+    not_found unless matching_company
+    @filtered_by = matching_company
+  end
```

Comment:
I just think this method does a lot of things and it seems not to be quite clear of its main purpose.

Let me try to explain, `check_company_subdomain` doesn't necessarily says that it is going to filter anything as saying here
```
@filtered_by = matching_company
```

IMHO `check_company_subdomain` should only checks for its domain, I would **suggest** something in this direction

```ruby
before_action :filter_by_company if allowed_subdomain?

def allowed_subdomain?
  subdomain = request.subdomain
  !subdomain.empty? && subdomain != 'www'
end

def filter_by_company
  @filter_by = matching_company or not_found
end

def matching_company
  Company.find_by(domain: request.subdomain)
end

def not_found
  raise ActionController::RoutingError, 'Not Found'
end
```

I just wanted to point that out, I've been starring at this method a while 😆 
What do you guys think? @samuelgrigolato @duduribeiro

Reply:
I totally agree with this 💯 ..

Reply:
Indeed this proposal results in a way more readable code, thanks! Just a question, though: Is  `allowed_subdomain?` the best name for that function? I mean, it is not like *no domain* or `www` are not allowed, they just won't trigger the company filtering feature. What about something like `company_subdomain?`? What do you both think?

Reply:
I'm with you on `company_subdomain?` 😃

Reply:
@fpgentil @duduribeiro I tried to apply this thread's suggestion as close as possible, but I couldn't find a way to make this work:

```ruby
before_action :filter_by_company if company_subdomain?

def company_subdomain?
  subdomain = request.subdomain
  !subdomain.empty? && subdomain != 'www'
end
```

This code gives me the following error:

> undefined method `company_subdomain?' for ApplicationController:Class

It seems like it is not possible to use class members directly in `before_action` declarations, I'm not sure though, so please enlighten me if that's incorrect :).

The way I ended up pushing to this PR is slightly different:

```ruby
before_action :filter_by_company

def filter_by_company
  if company_subdomain?
    @filter_by = matching_company || not_found
  end
end

def company_subdomain?
  subdomain = request.subdomain
  !subdomain.empty? && subdomain != 'www'
end
```

Let me know if you have any more suggestions.

Reply:
@samuelgrigolato `if` is an argument for the `before_action` and it expects a symbol with the same name of the method that you are calling. What happened in your code was that this if was a condition to check if it would add the `before_action` or not. And this happens at class level (not instance, like the `filter_by_company` method).

To fix it, you should change it to:

```
before_action :filter_by_company, if: :company_subdomain?
```

this is the same as:

```
before_action(:filter_by_company, if: :company_subdomain?)
```

so, the `if` is an argument to `before_action` and this before_action will always be present on the class and it will check the condition with the argument passed in parameter.

Reply:
@duduribeiro works perfectly! Thank you! I've just pushed a new version with this conditional in the `before_action` call instead of in the `filter_by_company` function.